extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -263,8 +263,10 @@ export abstract class ToolBase implements ITool {
 		);
 		this.version = this.getVersionFromOutput(commandOutput);
 		if (this.version) {
-			// discover and set the installationPath
-			await this.setInstallationPath();
+			if (this.autoInstallSupported) {
```

Comment:
good fix!

Reply:
If we found a version installed then there is nothing wrong with discovering its path and returning that. So I would do the fix to add discoveryCommand for Docker and make that part work for it. But what you have is better than current behavior so thank you very much for the fix.